packages/flutter/lib/src/cupertino/text_selection_toolbar_button.dart

```diff
@@ -104,17 +109,28 @@ class CupertinoTextSelectionToolbarButton extends StatelessWidget {
 
   @override
   Widget build(BuildContext context) {
+    final bool isDarkMode = MediaQuery.of(context).platformBrightness == Brightness.dark;
+    final Color backgroundColor = isDarkMode
+        ? _kToolbarBackgroundColorDark
+        : _kToolbarBackgroundColorLight;
+    final Color textColor = isDarkMode
+        ? CupertinoColors.white
+        : CupertinoColors.black;
+
     final Widget child = this.child ?? Text(
-      getButtonLabel(context, buttonItem!),
-      overflow: TextOverflow.ellipsis,
-      style: _kToolbarButtonFontStyle.copyWith(
-        color: onPressed != null ? CupertinoColors.white : CupertinoColors.inactiveGray,
-      ),
-    );
+       text ?? getButtonLabel(context, buttonItem!),
+       overflow: TextOverflow.ellipsis,
+       style: _kToolbarButtonFontStyle.copyWith(
+         color: onPressed != null
+             ? textColor
+             : CupertinoColors.inactiveGray,
+       ),
+     );
+
     return CupertinoButton(
       borderRadius: null,
-      color: _kToolbarBackgroundColor,
-      disabledColor: _kToolbarBackgroundColor,
+      color: backgroundColor,
```

Comment:
same comment as above. I think CupertinoButton resolves dynamic colors so you can just construct a dyanmic color and give it to the `CupertinoButton` here